test/e2etest/kubetest/kubetest_runner.go

```diff
@@ -0,0 +1,231 @@
+		kubtestArgs := createKubetestArgs(parallelTestsFocus, true, false, config.FlakeAttempts)
+		log.Info("Run kubetest in parallel way")
+		log.Infof("kubetest dump dir: %s", kubtestArgs.LogDir)
+		runKubetest(kubtestArgs)
+	}
+	if serialTestsFocus != "" {
+		kubtestArgs := createKubetestArgs(serialTestsFocus, false, false, config.FlakeAttempts)
+		log.Info("Run kubetest in serial way")
+		log.Infof("kubetest dump dir: %s", kubtestArgs.LogDir)
+		runKubetest(kubtestArgs)
+	}
+	return config.LogDir
+}
+
+func getFileContent(filepath string) string {
+	if file, err := os.Open(filepath); err != nil {
+		log.Fatal(err)
+	} else {
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			log.Fatal(err)
+		}
+		return string(b)
+	}
+	return ""
+}
+
+func createKubetestArgs(ginkgoFocus string, parallel, dryRun bool, flakeAttempts int) KubetestArgs {
+	ginkgoParallelArg := ""
+	if parallel {
+		ginkgoParallelArg = "--ginkgo-parallel"
+	}
+	if ginkgoFocus != "" {
+		ginkgoFocus = fmt.Sprintf("--ginkgo.focus=%s", ginkgoFocus)
+	}
+	logDir := filepath.Join(config.LogDir, strconv.FormatInt(time.Now().Unix(), 10))
+	_ = os.MkdirAll(logDir, os.FileMode(0777))
+	kubetestArgs := KubetestArgs{ShootConfigPath: config.ShootKubeconfigPath, GinkgoFocus: ginkgoFocus, DryRun: dryRun, LogDir: logDir, FlakeAttempts: flakeAttempts, Provider: Cloudprovider(config.CloudProvider), GinkgoParallel: ginkgoParallelArg}
+	return kubetestArgs
+}
+
+func runKubetest(args KubetestArgs) {
+	ginkgoArgs := fmt.Sprintf("--test_args=--ginkgo.flakeAttempts=%o --ginkgo.dryRun=%t %s", args.FlakeAttempts, args.DryRun, args.GinkgoFocus)
+	cmd := exec.Command("go", "run", "hack/e2e.go", "--", "--provider=skeleton", "--deployment=local", "--test", "--check-version-skew=false", args.GinkgoParallel, ginkgoArgs, fmt.Sprintf("--dump=%s", args.LogDir))
+	cmd.Dir = config.KubernetesPath
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	if log.GetLevel() == log.DebugLevel {
+		cmd.Stderr = &stderr
+	}
+	err := cmd.Run()
+
+	if log.GetLevel() == log.DebugLevel {
+		log.Info(fmt.Printf(stderr.String()))
+	}
+
+	//	Output our results
+	if out.String() != "" {
+		e2eLogFilePath := filepath.Join(args.LogDir, "e2e.log")
+		err := ioutil.WriteFile(e2eLogFilePath, out.Bytes(), 0644)
+		if err != nil {
+			log.Error(errors.Wrapf(err, "Couldn't write file %s", e2eLogFilePath))
+		}
+	}
+
+	// kubetest run fails if one of the testcases failes, therefore the execution is still successful and no err needs to be returned
+	if err != nil {
+		log.Error(errors.Wrapf(err, "kubetest run failed"))
+	} else {
+		log.Info("kubtest test run successful")
+	}
+}
+
+func escapeAndConcat(descFilePath string) (concatenatedParallelTests, concatenatedSerialTests string) {
+	var serialTestsBuffer bytes.Buffer
+	var parallelTestsBuffer bytes.Buffer
+
+	file, err := os.Open(descFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
```

Comment:
you need to be careful with the scanner. It can overflow if the descFile is big.
How big can is this file normally?